### ISSUE TITLE
GUI Training: SLEAP does not use non-GUI parameters when using training config as template

### ISSUE THREAD

**Issue by roomrys** [talmolab/sleap#1052]
# Bug description
Loading a training config in the GUI does not use the loaded training config's hidden parameters, i.e. parameters not displayed in the GUI.

## Expected behaviour

The loaded training config's parameters will be used as defaults for parameters which are not set in the GUI.

## Actual behaviour

Parameters set in GUI are used, but parameters not set in GUI have defaults determined by `TrainingJobConfig` class in code.

## Your personal set up

</details>

## How to reproduce

1. Go to Training Pipeline in GUI
2. Click on Model Configuration tab
3. Select model from drop-down.
4. Run

**Comment by roomrys** [comment]
# Relevant Code

- `run` calls `get_ever_head_config_data`
https://github.com/talmolab/sleap/blob/5093f6992e6214c0d528b7240331b99d0a89a62f/sleap/gui/learning/dialog.py#L581-L617

- `get_every_head_config_data` calls `make_training_config_from_key_val_dict`
https://github.com/talmolab/sleap/blob/5093f6992e6214c0d528b7240331b99d0a89a62f/sleap/gui/learning/dialog.py#L415-L446

- `make_training_config_from_key_val_dict` creates `TrainingJobConfig`
https://github.com/talmolab/sleap/blob/5093f6992e6214c0d528b7240331b99d0a89a62f/sleap/gui/learning/scopedkeydict.py#L177-L191

- `TrainingJobConfig` has defaults for values not specified in GUI (`outputs = OutputsConfig`)
https://github.com/talmolab/sleap/blob/5093f6992e6214c0d528b7240331b99d0a89a62f/sleap/nn/config/training_job.py#L41-L62

**Comment by roomrys** [comment]
This feature is now available in the (pre) release 1.3.0a0, to install, first [uninstall](https://sleap.ai/installation.html#upgrading-and-uninstalling) and then:
**`conda` (Windows/Linux/GPU)**:
```
conda create -y -n sleap -c sleap -c sleap/label/dev -c nvidia -c conda-forge sleap=1.3.0a0
```
**`pip` (any OS except Apple Silicon)**:
```
pip install sleap==1.3.0a0
```

**Warning:** This is a pre-release! Expect bugs and strange behavior when testing.
